lineapy/cli/cli.py

```diff
@@ -371,6 +381,148 @@ def setup_ipython_dir() -> None:
     os.environ["IPYTHONDIR"] = ipython_dir_name
 
 
+@linea_cli.group("annotate")
+def annotations():
+    """
+    The annotate command can be used to import custom annotation sources
+    into LineaPy. It can be used to add a source, list all sources,
+    delete a source, and validate all sources.
+    """
+    pass
+
+
+def validate_annotations_path(ctx, param, value: pathlib.Path):
+    if value.suffix != ".yaml":
+        raise click.BadParameter("path must be a yaml file")
+    return value
+
+
+@annotations.command("add")
+@click.argument(
+    "path",
+    type=click.Path(dir_okay=False, path_type=pathlib.Path),
+    callback=validate_annotations_path,
+)
+@click.option(
+    "--name",
+    "-n",
+    default=None,
+    help="What to name imported resource (do not include .yaml extension). Input file name is used as default",
+    type=str,
+)
+def add(path: pathlib.Path, name: str):
+    """
+    Import user-defined function annotations into Lineapy.
+
+    This command copies the yaml file whose path is provided by the user into the user's .lineapy directory to allow Lineapy to manage it.
+    """
+    # Sanitize input before using it as filename
+    name = name or path.stem
+    name = slugify(name)
+
+    annotate_folder = custom_annotations_folder()
+
+    # Path to copy destination in user's .lineapy directory
+    destination_file = annotate_folder / (
+        name + CUSTOM_ANNOTATIONS_EXTENSION_NAME
+    )
+    print(f"Creating annotation source at {destination_file}")
+
+    # Copy annotation file to destinatiion
+    try:
+        shutil.copyfile(path, destination_file)
+    except IOError as e:
+        logger.error(
+            f"Failed to copy file from {path} to {destination_file}.\n{str(e)}"
+        )
+        sys.exit(1)
+
+
+@annotations.command("list")
+def list():
+    """
+    Lists full paths to all imported annotation sources.
+    """
+    annotate_folder = custom_annotations_folder()
+    for annotation_path in annotate_folder.iterdir():
+        if (
+            annotation_path.is_file()
+            and re.match(CUSTOM_ANNOTATIONS_REGEX_MATCH, str(annotation_path))
+            is not None
+        ):
+            print(annotation_path)
+
+
+@annotations.command("delete")
+@click.argument("filename")
+def delete(filename: str):
+    """
+    Deletes imported annotation source.
+    """
+    # filename must end with .annotations.yaml
+    if re.match(CUSTOM_ANNOTATIONS_REGEX_MATCH, filename) is None:
+        logger.error(
+            f"Resource to delete must end with '{CUSTOM_ANNOTATIONS_EXTENSION_NAME}'"
+        )
+        sys.exit(1)
```

Comment:
this looks like a bad UX. The add takes in a name without any extensions but the delete requires the full file name? We should support deletes by the name or by the full filename when deleting (if we want to support full name at all). So if you don't find a match using your regex, you should probably try once again after appending the CUSTOM_ANNOTATIONS_EXTENSTION_NAME and if there are still no matches then log the error. Alternately, keep the UX the same as "add" where you only provide the name.